lib/warnless.h

```diff
@@ -60,6 +60,12 @@ int curlx_sktosi(curl_socket_t s);
 
 curl_socket_t curlx_sitosk(int i);
 
+#else  /* USE_WINSOCK */
+
+#define curlx_sktosi(s) ((int)(s))
+
+#define  curlx_sitosk(i) ((curl_socket_t)(i))
```

Comment:
Those two macros allow to use `curl_socket_t` to `int` and reverse castings safely on all platforms. And simplify coding.